### PR TITLE
feat(ST-121): fetch messages from db for specific classroom

### PR DESCRIPTION
**Pull request by ghost**
## Description of Change
- Updated message controller to fetch messages for specific classroom.
- Logic is implemented on message service layer

## Associated Ticket Link(s)

- https://trello.com/c/4M0049at

## Screenshots / Screen Recordings

N/A

### CODE REVIEW

File: src/messages/message.controller.ts [hunk 1]
```diff
@@ -24,4 +24,10 @@ export class MessagesController {
     const message: Message = await this.messagesService.createMessage(createMessageDto, user.id);
     return this.messageSerializer.serialize(message);
   }
+
+  @Get("classroom/:classroomId")
+  async getMessagesForClassroom(@Param("classroomId") classroomId: number) {
+    const messages = await this.messagesService.getMessagesByClassroom(classroomId);
```
Comment (ishmam-iqbal-sazim) [inline src/messages/message.controller.ts]: Let's add new line below

File: src/messages/message.controller.ts [hunk 2]
```diff
@@ -24,4 +24,10 @@ export class MessagesController {
     const message: Message = await this.messagesService.createMessage(createMessageDto, user.id);
     return this.messageSerializer.serialize(message);
   }
+
+  @Get("classroom/:classroomId")
```
Comment (ishmam-iqbal-sazim) [inline src/messages/message.controller.ts]: This should be `classrooms/:classroomId/messages`. The endpoint for the Post request should also be the same

File: src/messages/message.service.ts [hunk 3]
```diff
@@ -30,4 +30,9 @@ export class MessagesService {
 
     return this.messageRepository.createOne(messageData);
   }
+
+  async getMessagesByClassroom(classroomId: number): Promise<Message[]> {
+    const classroom = await this.classroomRepository.findOneOrFail(classroomId);
```
Comment (ishmam-iqbal-sazim) [inline src/messages/message.service.ts]:
```suggestion
    const classroom = await this.classroomsRepository.findOneOrFail(classroomId);
```
[NIT]

File: src/messages/messages.controller.ts [hunk 4]
```diff
@@ -25,4 +25,11 @@ export class MessagesController {
 
     return this.messagesSerializer.serialize(message);
   }
+
+  @Get("classroom/:classroomId/messages")
```
Comment (ishmam-iqbal-sazim) [inline src/messages/messages.controller.ts:29]:
```suggestion
  @Get("classrooms/:classroomId/messages")
```

Let's follow REST principles

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comment

**Review by ragib-hossain-sazim — APPROVED** [review]

**Pull request merged**